### PR TITLE
refactor: remove redundant `--disable-workload-identity`

### PR DESCRIPTION
**Pull request by bcho**
This pull request removed the redundant `--disable-workload-identity` flag from az aks create|update commands. Existing user can use `--enable-workload-identity False` to archive such functionality.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def load_arguments(self, _):
         c.argument('http_proxy_config')
         c.argument('enable_pod_security_policy', action='store_true')
         c.argument('enable_pod_identity', action='store_true')
-        c.argument('enable_workload_identity', arg_type=get_three_state_flag(), is_preview=True)
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/_params.py:286]: Is this expected to remove "is_preview=True"?
If so, shall we also remove it in line 367?

**Review by bingosummer — CHANGES_REQUESTED** [review]

**Review by bingosummer — APPROVED** [review]

**Comment by bcho** [comment]
> Please resolve CI conflicting files first.

Updated, PTAL @yonzhan

**Comment by zhoxing-ms** [comment]
Will removing parameter `--disable-workload-identity` directly have a great impact on users? 
If so, can we consider using `c.deprecate(hide=True)` to hide this parameter first? When most users migrate to parameter `--enable-workload-identity False`, then we remove it completely, so as to avoid big breaking change.

**Comment by bcho** [comment]
> Will removing parameter `--disable-workload-identity` directly have a great impact on users? If so, can we consider using `c.deprecate(hide=True)` to hide this parameter first? When most users migrate to parameter `--enable-workload-identity False`, then we remove it completely, so as to avoid big breaking change.

@zhoxing-ms 

Thanks for the concern. Removing these flags won't impact current user as in AKS, this feature is not being enabled yet, only a small group of testers have access to it. So, removing it won't cause breaking change.

**Comment by bcho** [comment]
@zhoxing-ms @yonzhan could you help merge this pr? Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**